### PR TITLE
Implement view for actuator endpoint

### LINKED ISSUE #1547: Support startup Actuator endpoint
It would be nice if the `startup` Actuator endpoint is supported with a human-friendly UI.

### PR DESCRIPTION
**Pull request by SteKoe**
closes #1547 

 * Navigate through event tree provided by startup actuator endpoint
 * Collapse/Expand whole event tree by using the checkbox
 * Persists current tree state and restores it on page reload in localStorage (when available)
 * Allows navigating to an event by adding `id=<id>` as a query param to the URL

image

### CODE REVIEW

**Comment by SteKoe** [comment]
Question: Do you think the link to the Startup overview beyond Logging is appropriate or should we move it to an own, dedicated menu item?

**Comment by joshiste** [comment]
What about showing the duration in milliseconds instead of seconds?

**Comment by SteKoe** [comment]
Switched to display milliseconds instead of seconds. A rounded value is shown in the column when hovering the value, an unrounded value is shown.

File: spring-boot-admin-server-ui/src/main/frontend/services/startup-actuator.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          parent = parent.startupStep.parent;
+          event.startupStep.depth++;
+        }
+
+        return event;
+      });
+
+    return new StartupActuatorEventTree(eventsForTree);
+  },
+  convertToMicroseconds: function (duration) {
```
Comment (Riggs333) [inline spring-boot-admin-server-ui/src/main/frontend/services/startup-actuator.js:112]: I wonder if this function catches the different values that the `duration` parameter can have.
The startup endpoint responds a serialized Java `Duration`: https://github.com/spring-projects/spring-boot/blob/769b5f0f03bd34e1d18d7b0ec54f0fcdacf0fd03/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/metrics/buffering/StartupTimeline.java#L76 

Should the function here handle `Duration`s that are of greater value. 
The format could be (see https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--):
`PT20.345S`  (equals 20.345 seconds)

Although the startup of an application is (or should) likely to be relatively fast. ;-)

Is it worth it to use some library for parsing the duration String? For example https://www.npmjs.com/package/iso8601-duration?

Regarding naming of the function: Is the returned valued of unit microseconds or milliseconds?

Comment (SteKoe) [reply spring-boot-admin-server-ui/src/main/frontend/services/startup-actuator.js]: Now supports up to any amount of weeks. I assume that a startup step does never take so long. The displayed value is in milliseconds.

Comment (Riggs333) [reply spring-boot-admin-server-ui/src/main/frontend/services/startup-actuator.js]: Sorry for being this late with my review, @SteKoe! 🙏🏻 
This looks already good to me.
The iso8601-duration library's example shows a `toSeconds` function:
```javascript
toSeconds( parse('PT1H30M10.5S')
```
Do you think that function can cover the converting part in our `toMilliseconds` function replacing all multiplications but the last?

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerInstanceWebClientConfiguration.java [hunk 2]
```diff
@@ -214,6 +214,12 @@ public LegacyEndpointConverter mappingsLegacyEndpointConverter() {
 			return LegacyEndpointConverters.mappings();
 		}
 
+		@Bean
```
Comment (Riggs333) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/config/AdminServerInstanceWebClientConfiguration.java:217]: I did not yet take a look what this bean is for, but it seems the travis build failed due to a test failure in `AdminServerInstanceWebClientConfigurationTest` where this new bean `startupLegacyEndpointConverter` is not yet expected.

**Pull request merged**